client/jetpack-cloud/sections/partner-portal/assign-license-step-progress/index.tsx

```diff
@@ -51,19 +49,25 @@ interface Props {
 	currentStep: StepKey;
 }
 
-export default function AssignLicenseStepProgress( { currentStep }: Props ): ReactElement {
+export default function AssignLicenseStepProgress( { currentStep }: Props ): ReactElement | null {
 	const translate = useTranslate();
-	const isAgency = useSelector( isAgencyUser );
-	const hasPaymentMethod = useSelector( hasValidPaymentMethod );
-	const paymentMethodRequired = isAgency && ! hasPaymentMethod;
+	const paymentMethodRequired = useSelector( isPaymentMethodRequired );
+	const sites = useSelector( getSites ).length;
 
-	const steps: Step[] = [
-		{ key: 'issueLicense', label: translate( 'Issue new license' ) },
-		...( paymentMethodRequired
-			? [ { key: 'addPaymentMethod', label: translate( 'Add Payment Method' ) } as Step ]
-			: [] ),
-		{ key: 'assignLicense', label: translate( 'Assign license' ) },
-	];
+	const steps: Step[] = [ { key: 'issueLicense', label: translate( 'Issue new license' ) } ];
+
+	if ( paymentMethodRequired ) {
+		steps.push( { key: 'addPaymentMethod', label: translate( 'Add Payment Method' ) } );
+	}
+
+	if ( sites > 0 ) {
+		steps.push( { key: 'assignLicense', label: translate( 'Assign license' ) } );
```

Comment:
Should we export the step keys from `client/jetpack-cloud/sections/partner-portal/assign-license-step-progress/index.tsx` so we define a constant here, instead of string values?